src/tasks/buildAndUpload.ts

```diff
@@ -48,12 +48,25 @@ import { releaseFiles } from "@dappnode/types";
 import { getImagePath } from "../utils/getImagePath.js";
 import { getLegacyImagePath } from "../utils/getLegacyImagePath.js";
 
+export interface BuildAndUploadOptions {
+  contentProvider: string;
+  uploadTo: UploadTo;
+  userTimeout?: string;
+  skipSave?: boolean;
+  skipUpload?: boolean;
+  requireGitData?: boolean;
+  deleteOldPins?: boolean;
+  composeFileName: string;
+  dir: string;
+  variantsDirPath?: string;
+  variantName?: string; // If variant name is undefined we are not using template mode
```

Comment:
why? this is confusing. It should be if `--template` use template with defaults or customs, if no `--template` then use single. Otherwise error

Reply:
Done!